### PR TITLE
applications: nrf_desktop: Non-volatile memory instead of flash in docs

### PR DESCRIPTION
**Pull request by MarekPieta**
Change switches to using non-volatile memory term instead of flash in the documentation. Other non-volatile memory may be used for the update image slot. Change also applies a few additional doc improvements.

Jira: NCSDK-24518

### CODE REVIEW

File: applications/nrf_desktop/doc/dfu.rst
Comment (kapi-no) [inline applications/nrf_desktop/doc/dfu.rst:1]: Can you also cover the DFU lock module, dfu_mcumgr, settings_loader and general nRF Desktop documentation (main Readme)?

Also, remember to add one changelog entry for this change (e.g. "Changed the flash memory term to the non-volatile memory term to generalize the nRF Desktop documentation")

Comment (MarekPieta) [reply applications/nrf_desktop/doc/dfu.rst:1]: Will be done under follow-up ticket: https://nordicsemi.atlassian.net/browse/NCSDK-24793

File: applications/nrf_desktop/doc/dfu.rst [hunk 1]
```diff
@@ -103,8 +103,8 @@ The DFU module implementation is centered around the transmission and the storag
 * `Partition preparation`_ - How the module prepares for receiving an image.
 
 The firmware transfer operation can also be carried out either by the :ref:`nrf_desktop_ble_smp` or :ref:`nrf_desktop_dfu_mcumgr`.
-The application user must not perform more than one firmware upgrade at a time.
-The modification of the data by multiple application modules would result in a broken image that would be rejected by the bootloader.
+Make sure that all enabled modules used for the firmware transfer operation use :ref:`nrf_desktop_dfu_lock` to synchronize non-volatile memory access.
+If a modules does not use the :ref:`nrf_desktop_dfu_lock`, the simultaneous modification of the data by the module and another application module would result in a broken image that would be rejected by the bootloader.
```
Comment (peknis) [inline applications/nrf_desktop/doc/dfu.rst:107]:
```suggestion
If a module does not use the :ref:`nrf_desktop_dfu_lock`, simultaneous modification of the data by the module and another application module results in a broken image that would be rejected by the bootloader.
```

File: applications/nrf_desktop/doc/dfu.rst [hunk 2]
```diff
@@ -173,7 +174,7 @@ start
 
 data
    Perform the set operation on this option to pass the chunk of the update image that should be stored at the current offset.
-   The data is initially buffered in RAM, and then it is written to the flash partition after a fetch operation is performed on the ``sync`` option.
+   The data is initially buffered in RAM, and then it is written to the non-volatile memory partition after a fetch operation is performed on the ``sync`` option.
```
Comment (peknis) [inline applications/nrf_desktop/doc/dfu.rst:177]:
```suggestion
   The data is initially buffered in RAM, and then written to the non-volatile memory partition after a fetch operation is performed on the ``sync`` option.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: applications/nrf_desktop/doc/dfu.rst (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/dfu.rst:107]: Changed to `may result` to make information more accurate

**Review by peknis — APPROVED** [review]

**Review by kapi-no — APPROVED** [review]

File: applications/nrf_desktop/doc/dfu.rst [hunk 3]
```diff
@@ -212,35 +213,37 @@ sync
...
 .. _dfu_bootloader_var:
 
 module_variant
    Perform the fetch operation on this option to get the information about the bootloader variant that is being used on the device.
 
 
-Writing data to flash
-=====================
+Writing data to non-volatile memory
+===================================
```
Comment (divipillai) [inline applications/nrf_desktop/doc/dfu.rst:225]: Shouldn't the dfu_lock.rst also be updated (line no: 24). We have mention of flash memory.

**Review by mkapala-nordic — APPROVED** [review]

File: applications/nrf_desktop/doc/dfu.rst (hunk 3 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/dfu.rst:225]: It will be updated in a separate PR.

**Review by divipillai — APPROVED** [review]

**Pull request merged**